packages/common/src/utils/http-utils.ts

```diff
@@ -24,9 +24,11 @@ export enum HttpMethods {
 
 export async function fetchJson(url: URL | string, opts: FetchOpts = {}): Promise<any> {
   if (opts.body) {
+    const headers = { 'Content-Type': 'application/json', ...opts.headers }
+
     Object.assign(opts, {
-      body: JSON.stringify(opts.body),
-      headers: { ...opts.headers, 'Content-Type': 'application/json' },
+      body: headers['Content-Type'] == 'application/json' ? JSON.stringify(opts.body) : opts.body,
```

Comment:
```suggestion
      body: headers['Content-Type'].includes('application/json') ? JSON.stringify(opts.body) : opts.body,
```
remember that content-type header can include encoding or other things at the end. like
```
Content-Type: text/html; charset=utf-8
Content-Type: application/json; charset=utf-8
Content-Type: multipart/form-data; boundary=something
```